(DOCSP-19418): Swift SDK Flexible Sync (#1540)
mongodb/docs-realm: source/sdk/swift/examples/configure-and-open-a-realm.txt

      .. literalinclude:: /examples/generated/code/start/OpenCloseRealm.codeblock.open-local-realm.m
         :language: objectivec

.. _ios-login-and-open-realm:

Open a Synced Realm
-------------------

The typical flow for opening a synced {+realm+} involves:

1. :ref:`Authenticating the user <ios-authenticate-users>`.
#. Creating a sync configuration.
#. Opening the user's synced {+realm+} with the configuration.

At authentication, we cache user credentials in a ``sync_metadata.realm`` 
file on device.

When you open a synced {+realm+} after authenticating, you can bypass the 
login flow and go directly to opening the synced {+realm+}, using the same 
sync configuration you already created.

With cached credentials, you can:

- Open a synced {+realm+} immediately with the data that is on the device.
  You can use this method offline or online.
- Open a synced {+realm+} after downloading changes from your {+app+}. 
  This requires the user to have an active internet connection.

> Synced Realms vs Local Realms

Reviewer: Shouldn't that be `vs.`?

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Synced {+realms+} differ from local {+client-database+} in a few ways:

- Synced {+realms+} attempt to sync changes with your backend {+app+},
  whereas local {+realms+} do not.
- Synced {+realms+} can be accessed by authenticated users, while local 

>   {+realms+} do not require any concept of users or authentication.

Reviewer: It feels like we're mixing two ideas in this, "do not require" and "any concept of". Maybe it should say:
```suggestion
  {+realms+} have no concept of users or authentication.
```

      <Classes/RLMRealmConfiguration.html#/c:objc(cs)RLMRealmConfiguration(py)syncConfiguration>` 
      object with the desired :objc-sdk:`partitionValue 
      <Classes/RLMSyncConfiguration.html#/c:objc(cs)RLMSyncConfiguration(py)partitionValue>` 
      to :objc-sdk:`+[RLMRealm realmWithConfiguration:error:]
      <Classes/RLMRealm.html#/c:objc(cs)RLMRealm(cm)realmWithConfiguration:error:>`.

      This opens a synced {+realm+} on the device. The {+realm+} 
      attempts to sync with your {+app+} in the background to check for changes 
      on the server, or upload changes that the user has made.

      .. literalinclude:: /examples/generated/code/start/Sync.codeblock.init-synced-realm.m
         :language: swift

.. _ios-flexible-sync-open-realm:

Open a Synced Realm with a Flexible Sync Configuration
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

When you use Flexible Sync, use the ``flexibleSyncConfiguration()``
to open a synced realm. 

.. code-block:: swift

   let app = App(id: YOUR_APP_ID_HERE)
   let user = try await app.login(credentials: Credentials.emailPassword(email: "email", password: "password"))
   let config = user.flexibleSyncConfiguration()

>    let realm = try await Realm.init(configuration: config, downloadBeforeOpen: .always)

Reviewer 2: can remove the `.init`, generally considered "unconventional"
